### PR TITLE
Adds an example data generator

### LINKED ISSUE #31: Built in sample data generators
We should include a few sample deephaven table generators to use for producing examples plots and documentation.

Sample data generator requirements:

1. Can be called of off dx without any other imports, ex. easy to call `sample_table = dx.sample_data()`
2. It doesn't use a separate thread or anything confusing to kill, ideally just deephaven table functions
3. It is deterministic, has the same data every time it is called (same random seed or generator?)
  a. like a `random_walk` or something where each rows random seed uses `ii` or something
5. It can be called repeatedly and returns a new table of the same sample data
6. It has columns covering all use cases that can be used to construct any sample plot
  a. or if it doesn't make sense for certain examples, a limited number of sample tables ex. `dx.sample_tree_data()` or 
  `dx.sample_data("tree")` or whatever, I am not dictating naming or how it's called
  b. maybe needs a separate table that covers multiple days, but only during business hours
  c. if we do have multiple tables, then maybe some shared keys would be good
9. The sample data looks visually pleasing (not just sin or linear, more like a random walk)
10. It starts with 1000 rows immediately upon being called, but also continues to infinitely tick additional data
11. It should be faux-stock looking data, but use invalid symbols or maybe something obviously fake theme like FISH, CAT, DOG, BIRD? (OOOO and 0000 are the only test symbols but look confusing)
12. It should have a variety of column types (maybe use learn deephaven enterprise tables as inspiration?)
  a. Timestamp = time table starts at 8am form January 1st, 1970 and increments in random 5sec intervals?
  b. string with 8 categories
  c. string with 4 categories
  d. price = double
  e. size = int
  f. a boolean with 90% true
 

Please edit this ticket as we narrow on requirements of how many tables and what the should look like.

### PR DESCRIPTION
**Pull request by dsmmcken**
Fixes #31

- adds a function so you can do `t = dx.example_data()` and get a ticking table back

I have the black formatter on, so it touched some extra lines.

### CODE REVIEW

File: src/deephaven/plot/express/data_generator/example_data.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+def example_data(ticking: bool = True, hours_of_data: int = 1) -> Table:
+    """Returns a Deephaven table containing a generated example data set.
+    Data is 1 hour of randomly generated (but deterministic) fictional
+    stock market data, and starts with the first 5 minutes of data
+    already initilized so your example plots won't start empty.
+
+    Contains the following columns:
+    - timestamp: a starting from the date deephaven.io was registered
+    - sym: a string represent a fictional stock symbol
```
Comment (jnumainville) [inline src/deephaven/plot/express/data_generator/example_data.py:25]: Very minor, but seems like timestamp is missing a word: "a starting time... " perhaps
And for sym, "a string representing..."

File: src/deephaven/plot/express/data_generator/example_data.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+                "dollars",
+                "side",
+                "SPet500",
+                "index",
+                "random = random_double",
+            ]
+        )
+    )
+
+    if ticking == True:
```
Comment (jnumainville) [inline src/deephaven/plot/express/data_generator/example_data.py:164]: The `== True` is redundant here
```suggestion
    if ticking:
```

File: src/deephaven/plot/express/data_generator/example_data.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+    def random_gauss(seed: int) -> float:
+        random.seed(seed)  # set seed once per row
+        return random.gauss(0, 1)
+
+    def random_list_sym(seed: int) -> str:
+        return random.choices(sym_list, sym_weights)[0]
+
+    def list_sym_index(sym: str) -> int:
+        return sym_list.index(sym)
```
Comment (jnumainville) [inline src/deephaven/plot/express/data_generator/example_data.py:71]: This is going to be doing a linear time lookup on sym_list every time the function is called, which although is unlikely to be a huge problem due to the short list, isn't necessary. 
There are a few ways to do this, but I'm thinking something like `sym_dict = { sym: i for i, v in enumerate(sym_list) }` below line 50 then line 71 becomes
`return sym_dict[sym]`

**Review by jnumainville — CHANGES_REQUESTED** [review]

File: src/deephaven/plot/express/data_generator/example_data.py (hunk 3 above)
Comment (dsmmcken) [reply src/deephaven/plot/express/data_generator/example_data.py]: good suggestion, I can skip the function too and just do that in the query string

**Review by jnumainville — APPROVED** [review]

**Pull request merged**